### DISCUSSION TITLE
Is it possible to negate pseudo selectors?

### DISCUSSION THREAD

**Discussion by LilyStilson** [AvaloniaUI/Avalonia#8357] [Q&A] [answered]
Let's say that I want a button that will have different `:pointerover` background color depending on whether it's focused or not.
Documentation doesn't say anything about pseudo selector negation, only class negation. I tried code bellow, it didn't worked for me, so I want to clarify if it is possible or not.

```XAML
<!-- Always applies -->
<Style Selector="Button:pointerover">
    <Setter Property="Background" Value="Blue" />
</Style>

<!-- Should override the :pointerover selector -->
<Style Selector="Button:pointerover:not(:focus)">
    <Setter Property="Background" Value="Red" />
</Style>
```

**Comment by maxkatz6** [comment, marked as answer]
> Documentation doesn't say anything about pseudo selector negation, only class negation

There is no difference in that context. Both can be negated.

> it didn't worked for me, so I want to clarify if it is possible or not.

You probably want to change background of the ContentPresenter inside of the button. See https://docs.avaloniaui.net/docs/styling/troubleshooting#selector-with-a-pseudoclass-doesnt-override-the-default

**Comment by Seikaijyu** [comment]
Try using the Button:pointerover /template/ ContentPresenter
